### PR TITLE
Allow work with CSE on user-mode (issue #36)

### LINKED ISSUE #36: Allow work with CSE on user-mode
New _Cloud Specific Extension_ commands:

**cio cloud-specific-extensions templates \<CMD>**:

- **list** > Lists List CSE templates
	GET	/v2/cse/templates

- **show** -> Shows CSE template
	GET	/v2/cse/templates/:template_id

- **import** -> Imports a CSE template
	POST /v2/cse/templates

- **update** -> Updates an existing CSE template identified by the given id
	PUT /v2/cse/templates/:template_id

- **list-deployments** -> List CSE deployments of a CSE template
	GET /v2/cse/templates/:template_id/deployments

- **delete** -> Deletes a CSE template
	DELETE /v2/cse/templates/:template_id

**cio cloud-specific-extensions deployments \<CMD>**:

- **list** > Lists List CSE deployments
	GET	/v2/cse/deployments

- **show** -> Shows CSE deployment
	GET	/v2/cse/deployments/:deployment_id

- **deploy** -> Deploys a new CSE deployment from CSE template
	POST /v2/cse/templates/:template_id/deployments

- **update** -> Updates an existing CSE deployment identified by the given id
	PUT /v2/cse/deployments/:deployment_id

- **delete** -> Deletes a CSE deployment
	DELETE /v2/cse/deployments/:deployment_id

# Expected Behavior

# Current Behavior

# Possible Solution

# Steps to Reproduce (for bugs)

    1.
    2.
    3.
    4.

# Context and environment

### PR DESCRIPTION
**Pull request by pcantera**
# Description
New _Cloud Specific Extension_ commands:
imagen

**cio cloud-specific-extensions templates <CMD>**:
imagen

- **list** > Lists List CSE templates
	GET	/v2/cse/templates

- **show** -> Shows CSE template
	GET	/v2/cse/templates/:template_id

- **import** -> Imports a CSE template
	POST /v2/cse/templates

- **update** -> Updates an existing CSE template identified by the given id
	PUT /v2/cse/templates/:template_id

- **list-deployments** -> List CSE deployments of a CSE template
	GET /v2/cse/templates/:template_id/deployments

- **delete** -> Deletes a CSE template
	DELETE /v2/cse/templates/:template_id

**cio cloud-specific-extensions deployments <CMD>**:
imagen

- **list** > Lists List CSE deployments
	GET	/v2/cse/deployments

- **show** -> Shows CSE deployment
	GET	/v2/cse/deployments/:deployment_id

- **deploy** -> Deploys a new CSE deployment from CSE template
	POST /v2/cse/templates/:template_id/deployments

- **update** -> Updates an existing CSE deployment identified by the given id
	PUT /v2/cse/deployments/:deployment_id

- **delete** -> Deletes a CSE deployment
	DELETE /v2/cse/deployments/:deployment_id

# Description

# Related Issue
Closes #36

# Motivation and Context

# How Has This Been Tested?

# Screenshots (if appropriate)

# Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

# Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: api/cloudspecificextension/deployments_api.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	data, status, err := csets.concertoService.Delete(fmt.Sprintf("/cse/deployments/%s", deploymentID))
+	if err != nil {
+		return err
+	}
+
+	if err = utils.CheckStandardStatus(status, data); err != nil {
+		return err
+	}
+
+	return nil
```
Comment (pbanos) [inline api/cloudspecificextension/deployments_api.go:120]: Deleting a CSE deployment just starts the undeploy process: you should return here a representation parsed from the response

File: api/cloudspecificextension/templates_api.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+	if err = json.Unmarshal(data, &template); err != nil {
+		return nil, err
+	}
+
+	return template, nil
+}
+
+// CreateTemplate creates a cloud specific extension template
+func (csets *CloudSpecificExtensionTemplateService) CreateTemplate(templateVector *map[string]interface{}) (template *types.CloudSpecificExtensionTemplate, err error) {
```
Comment (pbanos) [inline api/cloudspecificextension/templates_api.go:68]: templateVector is a bad name, what about templateParams?

File: cmd/cloud_specific_extension_deployments_cmd.go [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+	debugCmdFuncInfo(c)
+	svc, formatter := WireUpCloudSpecificExtensionDeployment(c)
+
+	checkRequiredFlags(c, []string{"id"}, formatter)
+	err := svc.DeleteDeployment(c.String("id"))
+	if err != nil {
+		formatter.PrintFatal("Couldn't delete CSE deployment", err)
+	}
+	return nil
+}
```
Comment (pbanos) [inline cmd/cloud_specific_extension_deployments_cmd.go:166]: You should probably show here the CSED as decommissioning

File: api/cloudspecificextension/templates_api.go (hunk 2 above)
Comment (pcantera) [reply api/cloudspecificextension/templates_api.go]: Done, additionally this should be done in more cases (added a note at issue #31 ).

**Pull request merged**